charts/zigbee2mqtt/templates/statefulset.yaml

```diff
@@ -40,6 +40,32 @@ spec:
       tolerations:
 {{ toYaml . | indent 8 }}
 {{- end }}
+      {{- if .Values.statefulset.writeableConfig }}
+      initContainers:
+        - name: yq
+          image: "{{ .Values.statefulset.configMigration.image.repository }}:{{ .Values.statefulset.configMigration.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          command: [ "/bin/sh" ]
+          securityContext:
+            runAsUser: {{ coalesce .Values.statefulset.securityContext.runAsUser "0" }}
+            runAsGroup: {{ coalesce .Values.statefulset.securityContext.runAsGroup "0" }}
+          args:
+            - -xc
+            - |
+              if [ ! -f /app/data/configuration.yaml ]; then
+                echo "Copying config, since destination does not exist.."
+                cp -v /config/configuration.yaml /app/data/configuration.yaml
+              else
+                echo "Found existing config. Merging values.."
+                yq eval-all --inplace '. as $item ireduce ({}; . * $item )' /app/data/configuration.yaml /config/configuration.yaml
```

Comment:
I am not sure how this script works, which file has preference? `/app/data/configuration.yaml` or `/config/configuration.yaml` ?
This is specially important since the migrations needs to be persisted, otherwise will get re-applied on every boot, and I suspect it would break the files.